alibi_detect/saving/saving.py

```diff
@@ -27,7 +26,10 @@
 C_REF_FILENAME = 'c_ref.npy'
 
 
-def save_detector(detector: Detector, filepath: Union[str, os.PathLike], legacy: bool = False) -> None:
+def save_detector(
+        detector: Union[BaseDetector, ConfigurableDetector],
```

Comment:
Replace `BaseDetector` with the `Detector` protocol?

Reply:
Agreed.